### DISCUSSION TITLE
How to make model follow mouse movement?

### DISCUSSION THREAD

**Discussion by ptsyhanok** [google/model-viewer#2743] [Q&A]
I need to make my model rotating while pointer is moved in viewport without pressing mouse button. 

Is it possible to make it? Maybe in the similar way we can link rotation to scroll, we can link it pointer movement?

Any help or ideas would me much appreciated!

**Comment by elalish** [comment]
Yes, just don't add the `camera-controls` attribute to MV (to stop our default action)  and change `cameraOrbit` inside of your mouse move event handler. It might help to look at our [pan](https://modelviewer.dev/examples/stagingandcameras/#panning) example.

**Reply by ptsyhanok** [reply]
Could you give me code if it, I need essentially model "look at" my pointer while it's moved. 

I'm very new to it, so it's a bit hard for me to navigate in the script. Thank you.

**Reply by NibeluR** [reply]
Same here. I'd like to make my 3D model follow cursor (without clicking anything) but I can't. I'm also new to modelviewer and JS. I wish there was a live example on web page or actual code. 🙏🙏🙏🙏

**Reply by elalish** [reply]
This is kind of jumping into the deep end if you're not familiar with JS. Start with some JS tutorials and looking at the source of other pages that do pointer-following first. Work your way up. You'll also need to think about the geometry and decide how you want to relate 2D mouse movement to 3D camera position, since that's also arbitrary.

**Reply by renanmalato** [reply]
I would like to now that too... is it possible to make object follow cursor hover?

**Comment by studiobythepine** [comment]
This code here works fine for my website, just a simple mouse tracker and modifying the cameraOrbit property... Edit the params to get the look you want

```
<html>
	 <model-viewer id="myModel" src='/3d/yourModel.here' ></model-viewer>
</html>

<script>
	function getMouse(e){
		return [
			Math.round(e.clientX / window.innerWidth * 100),
			Math.round(e.clientY / window.innerHeight * 90),
		]
	}

	document.onmousemove = function(e){
		const mouse = getMouse(e);
		const modelViewer = document.querySelector("#myModel")
		modelViewer.cameraOrbit = `${mouse[0]}deg ` + `${mouse[1] + 45}deg ` + `"110%"`
	}

</script>
```

**Reply by jayarbass** [reply]
this works, but its backwards/mirrored... when i move my mouse right, model looks left... and vice-versa...
